Improve blank line handling in block strings

softmoth opened the pull request:
Allow block strings to have blank lines that are not fully indented.
This is common in the wild, like:

    - foo: |
        Note there are not 4 spaces on the next line

        But this is still part of the block string.

As a side effect of fixing this, trailing newlines became significant,
so this also adds YAML's `|+` and `|-` chomp indicators to deal with
that.

Leont commented:
Looks good to me (patches with tests are awesome), could you squash them though?

softmoth commented:
I added another largish change which fixes handling of block strings that start with blank lines (and block strings consisting of only blank lines). And squashed all 3 into one commit.

softmoth commented:
And I added another small fix to add the explicit indentation indicator feature.

Leont commented on lib/YAMLish.pm6:
```diff
@@ -218,12 +221,39 @@ grammar Grammar {
+		]?
+		<.space>* <.comment>?
+
+		:my $max-blank;
```
I think initializing this to an empty string will simplify the rest of the logic.

softmoth replied:
Thanks, I have pushed that change.

Leont commented:
> And I added another small fix to add the explicit indentation indicator feature.

That's not a very small fix, but I suspect it can't really be made smaller :-/

softmoth commented:
Let me know if there's anything else that you see needs improved. I think I've simplified as much as I can without obfuscating things. Thanks for taking time to review!

Leont commented:
This has been solved in a completely different way in f624463dd3d3c4cd :-)
